L1Trigger/L1TMuonEndCapPhase2/src/DAQ/CSCTPConverter.cc

```diff
@@ -0,0 +1,181 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFConfiguration.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFConstants.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/EMTFContext.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/HostLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/SiteLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Data/ZoneLut.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/DAQ/SubsystemTags.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Utils/DebugUtils.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Utils/CSCUtils.h"
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/Utils/TPUtils.h"
+
+#include "L1Trigger/L1TMuonEndCapPhase2/interface/DAQ/CSCTPConverter.h"
+
+using namespace emtf::phase2;
+
+CSCTPConverter::CSCTPConverter(const EMTFContext& context, const int& endcap, const int& sector)
+    : context_(context), endcap_(endcap), sector_(sector) {
+  // Do Nothing
+}
+
+CSCTPConverter::~CSCTPConverter() {
+  // Do Nothing
+}
+
+void CSCTPConverter::convert(const TriggerPrimitive& tp, const TPInfo& tp_info, EMTFHit& hit) const {
```

Comment:
I'm not going to ask for this change necessarily, but would it be possible to instead just return a new hit derived from the old? Are the side-effects of modifying the hit necessary? The purpose of the function can be a little clearer if it has a return value instead of side effect modifications of the arguments.

This applies to most of these `Converter` type classes.

Reply:
Maybe this can be changed in a future iteration. For now I don't consider it to be super urgent, but I agree it may be worth doing. Secondly, we would have to check how this is donw in FW, since the goal of the emulator is to imitate the FW as best as possible.